### DISCUSSION TITLE
How do I download 15-minute HRRR data?

### DISCUSSION THREAD

**Discussion by jahanbani** [blaylockbk/Herbie#67] [Q&A] [answered]
Hello,
I was wondering if the 15-minute data can be downloaded using Herbie. If so, how is it possible?

Thank you very much for your help.

**Comment by blaylockbk** [comment, marked as answer]
Yes, you can download HRRR 15-min data with `product="subh"` (subh is for subhourly).

```python
from herbie.archive import Herbie
H = Herbie('2022-04-07 00:00', model='hrrr', product='subh', fxx=2)
```

- `fxx=0` is the analsyis
- `fxx=1` will return the variables at 15, 30, 45, and 60 minutes since the analysis.
- `fxx=2` will return the variables at 75, 90, 105, and 120 minutes since the analysis,
- etc.

---

### Subsetting

If you only want, for example, the temperature data, you can use the **searchString** argument to subset.

Reading the index file for a subset is a quick way to verify the searchString
```python
H = Herbie('2022-04-07 00:00', model='hrrr', product='subh', fxx=2)
H.read_idx(searchString="TMP")
```
![image](https://user-images.githubusercontent.com/6249613/162271927-0b22887f-16d9-41e0-980a-b9a375be082f.png)

The data can be read into xarray with
```python
ds = H.xarray(searchString="TMP")
```

When you do that, it seems that cfgrib doesn't want to read all the fields in the same Dataset (it returned a list of two Datasets), but you can combine them with
```python
import xarray as xr
ds_new = xr.concat(ds, dim='step', data_vars=['t2m'])
```
![image](https://user-images.githubusercontent.com/6249613/162271455-659f4c12-c4d1-478f-97cb-fc5656a61634.png)

**Comment by jahanbani** [comment]
@blaylockbk 
Thank you very much for your comments. Very helpful. 

I am trying to use HRRR to extract wind speeds (at 80 m)/solar potential to calculate the potential for renewable energies at each point.

I am ok with using historical data rather than predictions, but I wonder if historical data (based on observation) is included in HRRR.

What is the `searchString` for the net solar irradiance on the ground?

Thank you.

**Reply by blaylockbk** [reply]
1. The model analysis (also known as `fxx=0`, forecast lead time zero, or F00) is a blend of the HRRR model background (which is the forecast from the previous run) and recent observations (from aircraft, radar, radiosonde, mesonets, etc.). These observations are incorporated into the model with data assimilation.
2. **DSWRF:surface** is the downward short-wave radiation flux at the surface. You could use `searchString="DSWRF:surface"`.
    - If you look at the forecast files, you will see there are two different fields for DSWRF. One is the instantaneous solar flux at the surface at each 15-min interval, the other is the average for each 15-min interval. You can use regular expressions to match just the instantaneous fields `DSWRF:surface.*min fcst` or just the average values `DSWRF:surface.*ave fcst`

 
Similarly, you can use regular expression to match both the 80-m *u* and *v* wind components with `searchString="(?:U|V)GRD:80 m"`

Note that the model analysis is only available at hourly intervals. If you want 15-minute solar and wind data from HRRR, you would have to settle with using forecasted data. F00 is the analysis as I've said before, and F01 has 15, 30, 45, and 60 minute forecasts.

**Reply by jahanbani** [reply]
Thank you very much for Herbie and for answering my questions 😀
